src/force-latest-compat.jl

```diff
@@ -0,0 +1,54 @@
+using TOML
+
+function latest_compat(range::VersionRange)
+    lower = range.lower
+    upper = range.upper
+    t_lower = Int.(lower.t[1:lower.n])
+    t_upper = Int.(upper.t[1:upper.n])
+    # `t_lower` is a tuple of integers
+    # `t_upper` is a tuple of integers
+    # `t_lower` and `t_upper` do not necessarily have the same length
+    if VersionNumber(t_lower) > VersionNumber(t_upper)
+        return t_lower
+    else
+        return t_upper
+    end
+end
+
+function latest_compat(s::VersionSpec)
+    ts = latest_compat.(s.ranges)
+    # `ts` is a vector of tuples of integers
+    # the tuples in `ts` may have different lengths
+    vers = VersionNumber.(ts)
+    i = argmax(vers)
+    t = ts[i]
+    new_compat_entry = join(string.(t), ".")
+    return new_compat_entry
+end
+
+function latest_compat(compat_entry::AbstractString)
+    s = semver_spec(compat_entry)
+    return latest_compat(s)
+end
+
+function force_latest_compat(compat_old::AbstractDict{<:AbstractString, <:Any})
+    compat_new = Dict{String,Any}()
+    for (pkg_name, old_compat_entry) in pairs(compat_old)
+        compat_new[pkg_name] = latest_compat(old_compat_entry)
+    end
+    return compat_new
+end
+
+function force_latest_compat(filename::AbstractString)
```

Comment:
This function looks kinda pointless. We already have functions to read and write projects. What you want to do is just change the compat of an existing project. So the only thing that should be modified is https://github.com/JuliaLang/Pkg.jl/blob/8b0045daf928e55e06a4701d959caba2487bd63f/src/Types.jl#L210.

Reply:
Sure, we can get rid of the `force_latest_compat(filename::AbstractString)` method. However, we would just end up moving this code (the code to load the project from the `Project.toml` file) from this location to line 1443 of `src/Operations.jl`.

If you look at lines 1437 through 1443 of `src/Operations.jl`: the temp project that is going to be used for the test sandbox is in the form of a `Project.toml` file that has been written to disk at the filename `tmp_project`. This temp project does not exist in memory yet; it is only a project file on disk. So we will need to read it in from the file.